### PR TITLE
dev: fix: SDK-191: Access denied errors in some situation while using boto2

### PR DESCRIPTION
**Pull request by harshshah87**
We get access  denied errors in some situations for boto2. Adding a catch to make sure we retry in such situations.

### CODE REVIEW

File: qds_sdk/commands.py [hunk 1]
```diff
@@ -1316,7 +1316,18 @@ def _is_complete_data_available(bucket_paths, num_result_dir):
...
         log.info("Downloading file from %s" % s3_path)
         if delim is None:
-            key_instance.get_contents_to_file(fp)  # cb=_callback
+            try:
+                key_instance.get_contents_to_file(fp)  # cb=_callback
+            except boto.exception.S3ResponseError as e:
+                if (e.status == 403):
+                    # SDK-191, boto gives an error while fetching the objects using versions which happens by default
+                    # in the get_contents_to_file() api. So attempt one without specifying version.
+                    log.info("Access denied while fetching the s3 object. Retrying without specifying the version....")
```
Comment (sourabh912) [inline qds_sdk/commands.py]: nit: log.error ?

Comment (sourabh912) [inline qds_sdk/commands.py]: Also have we checked with AWS (or read in their sdk docs) of this behaviour ? This seems like a hack. Are we able to reproduce it in house ?

**Comment by msumit** [comment]
lgtm

**Comment by harshshah87** [comment]
Merged

**Pull request closed without merging**